R/tbl_json.R

```diff
@@ -185,33 +224,80 @@ is.tbl_json <- function(.x) inherits(.x, "tbl_json")
 #' 
 #' @return A nested list representing the JSON data
 #' 
-#' @rdname json_get
 #' @export
 json_get <- function(.data) {
   .data[["..JSON"]]
 }
 
+#' Make the JSON data a persistent column
+#' 
+#' Extract the raw JSON from a tbl_json object. Store it in a column. WARNING:
+#' column name collisions will be overwritten
+#' 
+#' @param .data A tbl_json object
+#' @param column_name Optional. The name of the output column (either as a
+#'   string or unquoted name). Default "json"
+#' 
+#' @return A tbl_json object with an added column containing the JSON data
+#' 
+#' @examples
+#' 
+#' tj <- as_tbl_json('{"a": "b"}')
+#' json_get_column(tj, my_json)
+#' 
+#' @export
+json_get_column <- function(.data, column_name = "json") {
+  # TODO: protect against name collisions somehow?
+  qnm <- rlang::enquo(column_name)
+  colnm <- rlang::as_name(qnm)
+  if ("json" %in% names(.data)) {
+    warning(paste0("Column `", colnm, "` already exists. It will be overwritten by `json_get_column()`"))
+  }
+  .data[[colnm]] <- json_get(.data)
+  return(.data)
+}
+
 #' Wrapper for extending dplyr verbs to tbl_json objects
 #' @param dplyr.verb a dplyr::verb such as filter, arrange
 #' @param generic character. The name of the generic
 #' @keywords internal
 wrap_dplyr_verb <- function(dplyr.verb, generic) {
 
   function(.data, ...) {
-
+    
+    json <- json_get(.data)
+    
     # Apply the transformation
-    y <- NextMethod(generic, .data)
+    if (generic %in% c("select")) {
+      # some generics need a tibble
```

Comment:
otherwise `select` will add `..JSON` as a grouping variable and warn (confusingly)